xcube/util/jsonschema.py

```diff
@@ -101,10 +101,26 @@ def validate_instance(self, instance: Any):
         # jsconschema needs extra packages installed to validate some formats;
         # if they are missing, the format check will be skipped silently. For
         # date-time format, strict_rfc3339 or rfc3339-validator is required.
+
+        schema = self.to_dict()
+
+        # By default, jsonschema only recognizes lists as arrays. Here we derive
+        # and use a custom validator which recognizes both lists and tuples as
+        # arrays.
+        base_validator = jsonschema.validators.validator_for(schema)
+        new_type_checker = \
```

Comment:
This wiring up here is not clear to me. Is it really that complicated to add a custom type check?

Reply:
I agree, it's rather involved compared to the old method, but this really is officially The Right Way according to [the JSON Schema documentation](https://python-jsonschema.readthedocs.io/en/stable/validate/#validating-with-additional-types), and I couldn't find a simpler one.

Reply:
Bah, I don't like that.